eventuals/task.h

```diff
@@ -19,12 +20,13 @@ namespace eventuals {
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename E_, typename From_, typename To_>
+template <typename E_, typename From_, typename To_, typename Catches_, typename Raises_>
 struct HeapTask final {
   struct Adaptor final {
     Adaptor(
         Callback<function_type_t<void, To_>>* start,
-        Callback<void(std::exception_ptr)>* fail,
+        Callback<void(
+            typename VariantOfStoppedAndErrors<Raises_>::type)>* fail,
```

Comment:
Ok, getting back to a previous comment I made, this should not be a `std::variant<Stopped, Raises_...>`? We have the `stop` callback, should this just be a `std::variant` of `Raises_`.

And to handle an empty `std::variant` we should use `function_type_t`. So we'll want something like:
```suggestion
        Callback<void(variant_of_type_and_tuple_t<std::monostate, Raises_>&&)>* fail,
```
Then in `type-traits.h`:
```cpp
template <typename Type, typename Tuple>
using variant_of_type_and_tuple_t = apply_tuple_types_t<
    std::variant, 
    tuple_types_concatenate_t<std::tuple<Type>, Tuple>>;
```